### PR TITLE
Alternative linkerscript

### PR DESCRIPTION
**Pull request by itzandroidtab**
This needs some more testing. Not sure if everything is working as intended. It does separate the .stack and .bss.

### CODE REVIEW

File: vendor/cortex/atmel/sam3xa/source/as_gcc/sam3x8e_sram.ld [hunk 1]
```diff
@@ -39,7 +39,7 @@ SEARCH_DIR(.)
 MEMORY
 {
   rom (rx)  : ORIGIN = 0x00080000, LENGTH = 0x00080000
-  ram (rwx) : ORIGIN = 0x20000000, LENGTH = 0x00018000
+  ram (rwx) : ORIGIN = 0x20070000, LENGTH = 0x00018000
```
Comment (itzandroidtab) [inline vendor/cortex/atmel/sam3xa/source/as_gcc/sam3x8e_sram.ld:42]: This was not necessary in the end

File: boot/linkerscript.ld [hunk 2]
```diff
@@ -42,22 +47,15 @@ OUTPUT_ARCH(arm);
...
+	rom (rx) : org = 0x00000000, len = 512k
+	ram (rwx) : org = 0x20070000, len = 96k
 }
 __rom_start = ORIGIN(rom);
 __rom_size = LENGTH(rom);
 __rom_end = __rom_start + __rom_size;
 __ram_start = ORIGIN(ram);
 __ram_size = LENGTH(ram);
 __ram_end = __ram_start + __ram_size;
-PROVIDE(__rom_start = __rom_start);
```
Comment (itzandroidtab) [inline boot/linkerscript.ld:55]: The provide's are not needed so i removed them from the linkerscript

File: boot/linkerscript.ld [hunk 3]
```diff
@@ -79,176 +77,157 @@ ENTRY( __reset_handler );
...
-*(.eh_frame); /* information used for stack unwinding during exception */
-/*
-
+	.text :
+	{
+		. = ALIGN(4);
+		__text_start = .;
+		KEEP(*(.vectors .vectors.*))
+		*(.text .text.* .gnu.linkonce.t.*); /* text segment */
+		*(.glue_7t .glue_7); /* glue arm to tumb code, glue tumb to arm code*/
```
Comment (LRstudentHU) [inline boot/linkerscript.ld]: Some spelling errors in the comments

File: boot/linkerscript.ld [hunk 4]
```diff
@@ -79,176 +77,157 @@ ENTRY( __reset_handler );
...
+		*(.rodata .rodata.* .gnu.linkonce.r.*); /* Constands, strings, etc */
+		*(.ARM.extab* .gnu.linkonce.armextab.*); /* exception unwinding information */
```
Comment (LRstudentHU) [inline boot/linkerscript.ld]: Do we need this, since exceptions are disabled?

Comment (itzandroidtab) [reply boot/linkerscript.ld]: It doesnt look like it. It isn't in the default linkerscript [arm page](http://infocenter.arm.com/help/index.jsp?topic=/com.arm.doc.dui0474h/BABEHEDA.html). But was in the linker script before and I didn't think of removing it.

**Comment by LRstudentHU** [comment]
Is this ready to merge?

**Pull request merged**